src/SeleniumLibrary/keywords/browsermanagement.py

```diff
@@ -123,23 +125,36 @@ def open_browser(self, url, browser='firefox', alias=None,
         Applying ``desired_capabilities`` argument also for local browser is
         new in SeleniumLibrary 3.1.
         """
+        def _resolve_browser_alias_to_browser_name(name):
+            return WebDriverCreator.browser_names.get(name.lower(), name)
+
         if is_truthy(remote_url):
             self.info("Opening browser '%s' to base url '%s' through "
                       "remote server at '%s'." % (browser, url, remote_url))
         else:
             self.info("Opening browser '%s' to base url '%s'." % (browser, url))
-        driver = self._make_driver(browser, desired_capabilities,
-                                   ff_profile_dir, remote_url)
-        driver = self._wrap_event_firing_webdriver(driver)
+
+        try:
+            driver, driver_index = self.ctx._get_driver(alias)
+            browser_name = _resolve_browser_alias_to_browser_name(browser)
+            if driver.name != browser_name:
+                self.warn("Multiple browsers with alias %s being used." %
+                          alias)
+                raise RuntimeError
+        except (RuntimeError, ValueError, TypeError):
+            driver = self._make_driver(browser, desired_capabilities,
+                                       ff_profile_dir, remote_url)
+            driver = self._wrap_event_firing_webdriver(driver)
+            driver_index = self.ctx.register_driver(driver, alias)
+
         try:
             driver.get(url)
         except Exception:
-            self.ctx.register_driver(driver, alias)
```

Comment:
register driver is done right after its created so i removed this and below one because with our without issues, the driver *is* registered, albeit the driver's index wont be returned...